### ISSUE TITLE
Eliminate version from URI (or make configurable)

### ISSUE THREAD

**Issue by tedyoung** [spotify/docker-client#43]
After many days spent trying to figure out why pulls from my local docker-registry weren't working, I tracked it down to the "v1.12" version string that's added to all API calls. For some reason, when doing a POST to /v1.12/image/create, the docker daemon times out trying to connect to the docker-registry (though, in fact, it never actually tries to connect).

Dropping the version string entirely makes everything work. So, my request is for either dropping the version string, which is probably too risky, or making the version string configurable.

If making it configurable is an acceptable approach, I can submit a pull request for that change.

**Comment by lordofthejars** [comment]
Any news on this change? I can send a PR if you want making version configurable. I think we need this to be able to connect to as much docker servers as possible.

**Comment by rculbertson** [comment]
docker-client supports only one version of the API, so specifying other versions will likely cause serialization/deserialization errors. Omitting the version causes docker to default to the latest version. If omitting the version fixed an issue, maybe docker's backwards compatibly was broken. I'm a little hesitant to make this change if it was a workaround for a bug in docker. Maybe we can see if a similar issue occurs again, and then dig into what the most appropriate fix is?

**Comment by davidxia** [comment]
The version string has been updated from `v1.12` to `v1.15` which corresponds to Docker 1.3. We've updated the [README](https://github.com/spotify/docker-client/blob/master/README.md) to explicitly say we've only tested against Docker 1.3.

Making the version string configurable without running tests against all possible versions might mislead users into thinking the client will work with whatever Docker API version they specify.
